lib/main.dart

```diff
@@ -18,15 +18,27 @@ import 'package:flutter/material.dart';
 import 'src/app.dart' deferred as app;
 
 void main() async {
-  // Initialize Firebase
   WidgetsFlutterBinding.ensureInitialized();
-  await Firebase.initializeApp();
 
-  final Future<void> loadedLibrary = await app.loadLibrary();
+  final loaded = Future.wait([
+    app.loadLibrary(),
+    Firebase.initializeApp(),
+  ]);
+
   runApp(
     FutureBuilder(
-      future: loadedLibrary,
-      builder: (snapshot, context) => app.FriendlyEatsApp(),
+      future: loaded,
+      builder: (_, snapshot) {
+        if (snapshot.hasError) {
+          print(snapshot.error.toString());
+        }
+
+        if (snapshot.connectionState == ConnectionState.done) {
+          return app.FriendlyEatsApp();
+        }
+
+        return Container(color: Colors.white);
+      },
```

Comment:
This seems to be working fine without error on launch

Reply:
Do you really need to check the connectionState? I thought .hasData would be enough, check the example: https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html#widgets.FutureBuilder.1